dbms/src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -760,6 +766,13 @@ void MergeTreeData::loadDataParts(bool skip_sanity_checks)
             if (startsWith(it.name(), "tmp"))
                 continue;
 
+            Poco::Path marker_path(it.path(), DELETE_ON_DESTROY_MARKER_PATH);
```

Comment:
This operation extremely dangerous. Need logging.

Reply:
Added logging and switched to detaching instead of removing.